src/languages/go.ts

```diff
@@ -571,7 +570,7 @@ export class GoVisitor extends DefaultVisitor<GoLanguageContext> {
     if (renderer.currentContext.isInterface) {
       const type = this.renderTypeNode(node.type, true, renderer);
       const getter = new OTree([
-        renderer.updateContext({ isExported: renderer.currentContext.isExported && isPublic(node) }).convert(node.name),
+        renderer.updateContext({ isExported: isPublic(node) }).convert(node.name),
```

Comment:
I see what this is doing but I'm not convinced that this is the right solution yet.

Since rosetta is lower casing things that should be upper cased, and given that in Go every upper case variable is automatically exported, the context's `isExported` must be wrongly set to `false` somewhere. This fix decides to ignore the current `isExported` value entirely, and instead go by `isPublic`; but that doesn't seem right because `public` in TS and `export` in TS are not the same thing. `export` is needed to make something publicly accessible, `public` alone is not sufficient, so ignoring it here feels wrong.

The only time we want to make these interface properties capitalized is when we have an `export`ed interface, like this:

```
export interface IFoo {
  readonly foo: string;
}
```

Given that this bug is happening because the current context's `isExported` is wrongly set to `false`, when this code processes `foo` the *previous* context's `isExported` must be already `false`. The previous context is from the previous node, which must be the `IFoo` here. `IFoo` is `export`ed, so how is the previous context false? Can you figure out why that's happening?